### PR TITLE
Rolling release

### PR DESCRIPTION
**Pull request by a-teammate**
see issue #37 

stuff open in the publishing pipeline:
- [x] bump npm version via travis
- [x] set NPM_TOKEN in travis and appveyor as environment variables
- [x] ~remove media and flex submodules~ (not needed for this PR: its not in the package)

Summary:

This PR implements the automatic releasing of github releases each time we push to master.
(Thanks to GitHub we restrict pushes to master: each PR must have appveyor and travis running through)

additional info to https://github.com/inexor-game/code/issues/37#issuecomment-305863502:

We increment the patch version, based on the latest tag.
So each time we close a PR the newest commit gets tagged.

PR 1: 0.8.1-alpha
PR 2: 0.8.2-alpha
PR 3. 0.8.3-alpha
manually release by tagging 0.9: 0.9.0-alpha
PR 4: 0.9.1-alpha
PR 5: 0.9.2-alpha ...

So if we want to go for `0.9.x-alpha` we tag a commit with `0.9.0-alpha` manually.

Another note: we do not publish 32 bit packages currently (just out of lazyness)
so the NPM packages created are named `@inexorgame/inexor-core-windows64` and `@inexorgame/inexor-core-linux64`
@Fohlen the ZIP files pushed to github have different platform encodings.
if you want we can probably force windows64 and linux32 instead of win64 and i686 for them as well.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "@inexor-game/core-windows64",
```
Comment (aschaeffer) [inline package.json]: Does this approach scale on multiple platforms / archs?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "@inexor-game/core-windows64",
-  "version": "0.5.0",
+  "name": "@inexor-game/core-PLATFORM_PLACEHOLDER",
```
Comment (aschaeffer) [inline package.json]: Ok, ignore my last comment :-)

**Comment by a-teammate** [comment]
last review required by @Croydon:

If you generate a NPM token for me for "inexorgame" we can merge the rolling release PR
(we need to set NPM_TOKEN in travis and appveyor)

**Comment by Croydon** [comment]
I still feel like this isn't the most mature, stable and long-lasting way we can come up with, but go ahead for now.

**Comment by a-teammate** [comment]
Your feelings aside I still need the token ;D

**Comment by Croydon** [comment]
I gave you access to the NPM org before I even wrote the comment, didn't you get some notification?

**Comment by Fohlen** [comment]
NPM doesn't send notifications when you add users to institution accounts. They explicitely note that in the docs.

**Comment by a-teammate** [comment]
ah okay, thanks :)

**Pull request merged**